cloud/scope/powervs_cluster.go

```diff
@@ -882,37 +900,110 @@ func (s *PowerVSClusterScope) ReconcileNetwork() (bool, error) {
 	return false, nil
 }
 
-// checkNetwork checks if network exists in cloud with given name or ID mentioned in spec.
+// checkNetwork checks if network exists in cloud with given network's ID or name or dhcp server's ID or name mentioned in spec.
+// Priority order: Network ID - Network name - DHCP server ID - DHCP server name - Network with cluster name
 func (s *PowerVSClusterScope) checkNetwork() (*string, error) {
+	checkNetworkByID := func(networkID string) (*string, error) {
+		s.V(3).Info("Checking if PowerVS network exists in IBM Cloud with ID", "networkID", networkID)
+		network, err := s.IBMPowerVSClient.GetNetworkByID(networkID)
+		if err != nil {
+			return nil, err
+		}
+		return network.NetworkID, nil
+	}
+
 	if s.IBMPowerVSCluster.Spec.Network.ID != nil {
-		s.V(3).Info("Checking if PowerVS network exists in IBM Cloud with ID", "networkID", *s.IBMPowerVSCluster.Spec.Network.ID)
-		network, err := s.IBMPowerVSClient.GetNetworkByID(*s.IBMPowerVSCluster.Spec.Network.ID)
+		networkID, err := checkNetworkByID(*s.IBMPowerVSCluster.Spec.Network.ID)
 		if err != nil {
 			return nil, err
 		}
+		if err := s.verifyDHCPServerRespectToNetwork(*networkID); err != nil {
```

Comment:
I think this forces the user to set the dhcp network only. so we should update the api spec to mention it.
Also never tried what happens if we pass public network id here, Not relaying on DHCP.

Reply:
Yes. I think we should use DHCP server only, during boot, CoreOS expects the ip from a DHCP server. I hope the case is same for vanilla capi cluster as well.
Shall I modify the API spec as part of this PR?

Reply:
We use Centos images in normal capi flow as well. 
@mkumatag whats your thought. Should we make this restriction or do you foresee any other use cases.

Reply:
nope, we should allow both dhcp based and regular public networks.

Reply:
Ok, current code designed to handle only DHCP network. @Karthik-K-N We can create a story for this to handle later I believe. Wdyt?

Reply:
Yeah, That can be done.